buildSrc/src/main/kotlin/GenerateCompletionScriptsTask.kt

```diff
@@ -243,15 +277,61 @@ abstract class GenerateCompletionScriptsTask : DefaultTask() {
             }
     }
 
-    private fun generatePropertiesOpts(allPropertyNames: List<CliOption>): String {
-        return allPropertyNames
+    private fun generatePropertiesOpts(allPropertyNames: List<CliOption>) =
+        allPropertyNames
             .filter { it.oneDashOption != null }
             .joinToString("\n") {
-            val paddingOption = "-${it.oneDashOption!!.padEnd(40)}"
-            "$paddingOption - ${it.description?.lineSequence()?.first() ?: ""}"
+                val paddingOption = "-${it.oneDashOption!!.padEnd(40)}"
+                "$paddingOption - ${it.description?.lineSequence()?.first() ?: ""}"
+            }
+
+    private fun extractWrapperOptions(): List<TaskOptionDescriptor> {
+        return TaskOptionExtractor.extractTaskOptions("org.gradle.api.tasks.wrapper.Wrapper")
+    }
+
+    private fun extractTestOptions() =
+        TaskOptionExtractor.extractTaskOptions(
+            "org.gradle.api.tasks.testing.Test",
+            "org.gradle.api.tasks.testing.AbstractTestTask"
+        )
+
+    private fun extractInitOptions() =
+        TaskOptionExtractor.extractTaskOptions("org.gradle.buildinit.tasks.InitBuild")
+
+    private fun extractDependenciesOptions(): List<TaskOptionDescriptor> {
+        val options = TaskOptionExtractor.extractTaskOptions(
+            "org.gradle.api.tasks.diagnostics.DependencyReportTask",
+            "org.gradle.api.tasks.diagnostics.AbstractDependencyReportTask"
+        )
+
+        return addCustomCompletionFunctionForConfiguration(options)
+    }
+
+    private fun addCustomCompletionFunctionForConfiguration(options: List<TaskOptionDescriptor>) =
+        options.map { option ->
+            if (option.optionName == "configuration") {
+                option.copy(completionFunction = ":dependency configuration:_gradle_dependency_configurations")
+            } else {
+                option
+            }
```

Comment:
💅 This can be simplified

```suggestion
        options.map { option ->
            when(option.optionName) {
                "configuration" -> option.copy(completionFunction = ":dependency configuration:_gradle_dependency_configurations")
                else -> option
            }
        }
```